unified_planning/io/pddl_reader.py

```diff
@@ -878,27 +625,33 @@ def _parse_subtasks(
                 f"Could not parse the subtasks list: {e}, from line: {start_line}, col {start_col} to line: {end_line}, col {end_col}"
             )
 
-    def _check_if_object_type_is_needed(self, domain_res) -> bool:
-        for p in domain_res.get("predicates", []):
+    def _check_if_object_type_is_needed(self, domain_res, grammar: PDDLGrammar) -> bool:
+        empty_parse_res = ParseResults()
+        predicates = grammar.predicates[0][0] if grammar.predicates else empty_parse_res
```

Comment:
This indexing is **very** ugly. Can't we have a grammar property instead?

Reply:
Managed to clan it up by removing obsolete `Group` in the grammar.